### PR TITLE
ArmPkg/MdePkg: ArmPkg,MdePkg: Move ArmLib.h to MdePkg

### PR DESCRIPTION
**Pull request by pierregondois**
ArmPkg/MdePkg: ArmPkg,MdePkg: Move ArmLib.h to MdePkg

To allow the SecurityPkg to use functions to check arm cryptographic
instruction support and to avoid having the SecurityPkg relying on
another package than the MdePkg, move the ArmLib to the MdePkg [6].

More generally, this is a step for [5]. Indeed, the ArmPkg describes
industry standard information. Thus is should belong to the MdePkg.

Create a PR with [2] and [3] to move the ArmLib.h interface to the
MdePkg.

---------

[1]
[RFC PATCH 1/1] ArmPkg,MdePkg: move ArmLib.h to MdePkg
https://edk2.groups.io/g/devel/message/111566

[2]
[PATCH 0/2] ArmPkg/MdePkg: Move Chipset/* files to MdePkg
https://edk2.groups.io/g/devel/message/116673

[3]
[PATCH v2 0/2] ArmPkg/MdePkg: Move Chipset/* files to MdePkg
https://edk2.groups.io/g/devel/message/116759

[4]
[PATCH v5 0/6] CryptoPkg: Enable Openssl native instruction support for AARCH64
https://edk2.groups.io/g/devel/topic/patch_v5_2_6/102725178

[5]
BZ: https://bugzilla.tianocore.org/show_bug.cgi?id=4121

[6]
https://edk2.groups.io/g/devel/topic/patch_v5_2_6/102725178

### CODE REVIEW

**Comment by pierregondois** [comment]
CI fails due to:
`The commit adds/modifies files in multiple packages: ...`

**Comment by ardbiesheuvel** [comment]
> CI fails due to: `The commit adds/modifies files in multiple packages: ...`

Add the following to the commit log

Continuous-integration-options: PatchCheck.ignore-multi-package

**Comment by leiflindholm** [comment]
I approve this series. But we need a nod from the MdePkg maintainers as well. @mdkinney, @lgao4, could one of you set the approved flag if you're happy for this to go in?

**Review by ardbiesheuvel — APPROVED** [review]

**Review by mdkinney — APPROVED** [review]

**Review by lgao4 — APPROVED** [review]

**Pull request merged**
